zulip/zulip/__init__.py

```diff
@@ -688,6 +688,23 @@ def get_messages(self, message_filters):
             request=message_filters
         )
 
+    def get_messages_matching_narrow(self, request):
```

Comment:
I think this should be called `check_messages_match_narrow` -- the purpose is to determine whether a set of messages appear in a narrow (E.g. to determine whether to add a new message that just arrived to an open `search` narrow in the webapp).